public/js/NeighbourPlot.js

```diff
@@ -66,109 +66,218 @@ function NeighbourPlot(sampleData, element) {
+    var xMargin = (xMinMax[1]-xMinMax[0])*this.axisMargin;
+    var yMargin = (yMinMax[1]-yMinMax[0])*this.axisMargin;
+
+    if(!this.xScale) {
+        this.xScale = d3.scale.linear()
+            .range([0, this.width]);
+    }
+    this.xScale.domain([xMinMax[0]-xMargin, xMinMax[1]+xMargin]);
+
+    if(!this.yScale) {
+        this.yScale = d3.scale.linear()
+            .range([this.height, 0]);
+    }
+    this.yScale.domain([yMinMax[0]-yMargin, yMinMax[1]+yMargin]);
 
-    var xMin = d3.min(this.sampleData, function(d) { return d.pca_vector[0]; });
-    var yMin = d3.min(this.sampleData, function(d) { return d.pca_vector[1]; });
-    var xMax = d3.max(this.sampleData, function(d) { return d.pca_vector[0]; });
-    var yMax = d3.max(this.sampleData, function(d) { return d.pca_vector[1]; });
+};
 
-    // setup scales and axis
-    self.xScale = d3.scale.linear()
-        .domain([xMin-1, xMax+1])
-        .range([0, self.width]);
+/**
+ * drawAxis: Draw the plot axis.
+ *
+ * If the axis have not been drawn yet, they will be drawn. If the axis already
+ * exist, they are updated with an animation.
+ *
+ * @this {NeighbourPlot}
+ */
+NeighbourPlot.prototype.drawAxis = function() {
     var xAxis = d3.svg.axis()
-        .scale(self.xScale)
-        .ticks(self.xAxisTicks)
+        .scale(this.xScale)
+        .ticks(this.xAxisTicks)
         .orient('bottom');
 
-    self.yScale = d3.scale.linear()
-        .domain([yMin-1, yMax+1])
-        .range([self.height, 0]);
     var yAxis = d3.svg.axis()
-        .scale(self.yScale)
-        .ticks(self.yAxisTicks)
+        .scale(this.yScale)
+        .ticks(this.yAxisTicks)
         .orient('left');
 
-    // tooltip function
-    self.tip = d3.tip()
-        .attr('class', 'd3-tip')
-        .offset([-10, 0])
-        .html(function(d) {
-            return "<p><strong>ID: </strong>" + d._id + "</span></p>" +
-                "<p><strong>Gene: </strong>" + d.gene_name + "</span></p>";
-        });
+    // append containers for the x and y axis if they have not been defined yet
+    if(!this.xAxisSvg) {
+        this.xAxisSvg = this.svg.append('g')
+            .attr('class', 'x axis')
+            .attr('transform', 'translate(0,' + this.height + ')');
+    }
+
+    if(!this.yAxisSvg) {
+        this.yAxisSvg = this.svg.append('g')
+            .attr('class', 'y axis');
+    }
 
-    // setup canvas
-    self.svg = d3.select(this.element).append('svg')
-        .attr('width', self.fullWidth)
-        .attr('height', self.fullHeight)
-        .append('g')
-        .attr('transform', 'translate(' + self.margin.left + ',' + self.margin.top + ')')
-        .call(self.tip);
-
-    // append axis
-    self.svg.append('g')
-        .attr('class', 'x axis')
-        .attr('transform', 'translate(0,' + self.height + ')')
+    // apply the axis to the containers, an animation is applied when they're being updated
+    this.xAxisSvg
+        .transition()
+        .duration(750)
```

Comment:
I like this idea! Maybe it could be just be a JSON file where each key has the settings for some part of the UI.

That way plot behaviour/appearance can be tweaked quite easily from one location, or even adjusted by users if that's something we want to do much later down the track.